src/box/lua/config/annotations.lua

```diff
@@ -0,0 +1,2162 @@
+    well as logical operators (`||`, `&&`) and parentheses for grouping.
+]])
+
+-- DM
+-- SM
+M['conditional.*.*'] = toline([[
+    `conditional.if.*`: Defines a specific condition and the configuration
+    options applied when the condition is met. Each condition compares the
+    `tarantool_version` using operators (`<`, `>`, `<=`, `>=`, `==`, `!=`)
+    and supports defining labels or other options that are applied only if
+    the condition holds true.
+]])
+
+-- }}} conditional configuration
+
+-- {{{ config configuration
+
+M['config'] = toline([[
+    The `config` section defines various parameters related to
+    centralized configuration.
+]])
+
+-- {{{ config.context configuration
+
+M['config.context'] = toline([[
+    Specify how to load settings from external storage. For example,
+    this option can be used to load passwords from safe storage. You can
+    find examples in the Loading secrets from safe storage section.
+]])
+
+M['config.context.*'] = toline([[
+    The name of an entity that identifies a configuration value to load.
+]])
+
+M['config.context.*.env'] = toline([[
+    The name of an environment variable to load a configuration value from.
+    To load a configuration value from an environment variable, set
+    `config.context.<name>.from` to `env`.
+]])
+
+M['config.context.*.file'] = toline([[
+    The path to a file to load a configuration value from.To load a
+    configuration value from a file, set `config.context.<name>.from` to `file`.
+]])
+
+M['config.context.*.from'] = toline([[
+    The type of storage to load a configuration value from.
+    There are the following storage types:
+    `file`: load a configuration value from a file. In this case, you
+    need to specify the path to the file using `config.context.<name>.file`,
+    `env`: load a configuration value from an environment variable.
+    In this case, specify the environment variable name using
+    `config.context.<name>.env`.
+]])
+
+M['config.context.*.rstrip'] = toline([[
+    (Optional) Whether to strip whitespace characters and newlines
+    from the end of data.
+]])
+
+-- }}} config.context configuration
+
+-- {{{ config.etcd configuration
+
+M['config.etcd'] = toline([[
+    (EE) This section describes options related to providing connection
+    settings to a centralized etcd-based storage. If `replication.failover`
+    is set to `supervised`, Tarantool also uses etcd to maintain the state
+    of failover coordinators.
+]])
+
+M['config.etcd.endpoints'] = toline([[
+    The list of endpoints used to access an etcd server.
+]])
+
+M['config.etcd.endpoints.*'] = toline([[
+    etcd endpoint name. For example: `http://localhost:2379`.
+]])
+
+-- DM
+-- SM
+M['config.etcd.http'] = 'HTTP request configuration.'
+
+-- DM
+-- SM
+M['config.etcd.http.request'] = toline([[
+    Option for the HTTP request configuration. Contains settings for HTTP
+    requests, including request timeouts and socket connections.
+]])
```

Comment:
socket != unix socket.

If you want to lean on some existing documentation regarding these options, check out the etcd-client [documentation](https://tarantool.github.io/etcd-client/modules/etcd-client.html#new).

Reply:
Replaced with description from etcd-client docs.